libraries/DataFlash/LogStructure.h

```diff
@@ -34,6 +36,93 @@ struct PACKED log_Format {
     char labels[64];
 };
 
+struct PACKED log_Unit {
+    LOG_PACKET_HEADER;
+    uint64_t time_us;
+    char type;
+    char unit[64]; // you know, this might be overkill...
+};
+
+struct PACKED log_Format_Multiplier {
+    LOG_PACKET_HEADER;
+    uint64_t time_us;
+    char type;
+    double multiplier;
+};
+
+struct PACKED log_Format_Units {
+    LOG_PACKET_HEADER;
+    uint64_t time_us;
+    uint8_t format_type;
+    char units[16];
+    char multipliers[16];
+};
+
+struct UnitStructure {
+    const char ID;
+    const char unit[64];
+};
+
+struct MultiplierStructure {
+    const char ID;
+    const double multiplier;
+};
+
+// all units here should be base units
+// This does mean battery capacity is here as "amp*second"
+const struct UnitStructure log_Units[] = {
+    { '-', "" },    // no units e.g. Pi, or a string
+    { '?', "UNKNOWN" },    // Units which haven't been worked out yet....
+    { 'A', "amp" },
+    { 'd', "degrees" }, // of the angular variety, -180 to 180
+    { 'b', "bytes" },
+    { 'k', "degrees/second" },
+    { 'e', "degrees/second/second" },
+    { 'E', "radians/second" },
+    { 'G', "gauss" },
+    { 'h', "degheading" }, // 0.? to 359.?
+    { 'i', "amp*seconds" },
+    // { 'l', "litres" },
+    { 'L', "radians/second/second" },
+    { 'm', "metres" },
```

Comment:
meters

Reply:
We're using SI units here.  These are not for display purposes - that's the client's job.